/* ===> 1 <=== */

/*
 * successPromise должен зарезолвить число 67
 * Ответьте себе на вопрос, какой тип данных имеет переменная successPromise -
 * - object;
 */

const successPromise = new Promise((resolve) => {
  resolve(67);
});
console.log(successPromise);

/*
 * допишите обработчик успешного промиса (аргументы и тело ф-ции onSuccess)
 * чтобы в консоль вывелся квадрат числа, которое резолвиться из successPromise
 */

successPromise.then(function onSuccess(number) {
  console.log(number * number);
});

/*
 * в обработчике ошибок промиса (ф-ция onError внутри .catch()) выведите в консоль текст 'I am an error'
 * текст вывелся в консоль?
 * подумайте почему - because we didn`t have a mistake in our code
 * it cannot be amistake in a simple data type as number
 */
successPromise.catch(function onError() {
  console.log("I am an error");
});
